Notepad.Extensions.Logging/NotepadLogger.cs

```diff
@@ -82,30 +84,33 @@ void WriteMessage(LogLevel logLevel, int eventId, string message, Exception ex)
             _ => throw new ArgumentOutOfRangeException(nameof(logLevel)),
         };
 
-        static void WriteToNotepad(string message)
+        void WriteToNotepad(string message)
         {
             IntPtr hwnd = FindNotepadWindow();
+
+            if (hwnd.Equals(IntPtr.Zero))
+            {
+                return;
+            }
+
             IntPtr edit = NativeMethods.FindWindowEx(hwnd, IntPtr.Zero, "EDIT", null);
             NativeMethods.SendMessage(edit, NativeMethods.EM_REPLACESEL, (IntPtr)1, message);
         }
 
-        static IntPtr FindNotepadWindow()
+        IntPtr FindNotepadWindow()
         {
             IntPtr hwnd;
-            
-            hwnd = NativeMethods.FindWindow(null, "Untitled - Notepad");
-            if (hwnd != IntPtr.Zero)
-            {
-                return hwnd;
-            }
 
-            hwnd = NativeMethods.FindWindow(null, "*Untitled - Notepad");
-            if (hwnd != IntPtr.Zero)
+            hwnd = NativeMethods.FindWindow(null, windowName);
+            if (hwnd.Equals(IntPtr.Zero))
             {
-                return hwnd;
+                // when the file changes, notepad changes the name to "* Window Name", so later created loggers cannot find window
+                var builder = stringBuilderPool.Get();
+                builder.Append("*").Append(windowName);
```

Comment:
This seems overly excessive for every log call. You could create the alternative window name once in the constructor and cache it as well as the `windowName` itself.

Reply:
I also have a WIP that uses EnumWindows rather than FindWindow, which might be more efficient.

Reply:
This has been pushed to accommodate Notepad++ support. It should be easier now to match a given window name.